### DISCUSSION TITLE
Can the RESTful API of Microsoft Identity for bearer tokens be scaffolded ?

### DISCUSSION THREAD

**Discussion by LucianoCBarilari** [dotnet/aspnetcore#59662] [Q&A]
I created a backend Web API in Visual Studio using Microsoft Identity with bearer token authentication, but without a UI. I want to customize the default Identity endpoints, but I haven’t found a way to scaffold them. I tried scaffolding the UI following Microsoft’s documentation, but it didn’t affect the endpoints.

**Comment by engineering87** [comment]
Hello @LucianoCBarilari,
If you’re using .NET 8 with MapIdentityApi<TUser>(), the default JSON endpoints are registered there. There’s no scaffolding for those, so you need to copy the MapIdentityApi source from the ASP.NET Core repo, modify it, and map your own version.
If you’re using JWT with endpoints you wrote yourself, just edit your controllers or minimal API methods directly.
